backend/pkg/redpanda/service.go

```diff
@@ -66,26 +67,65 @@ func NewService(cfg config.Redpanda, logger *zap.Logger) (*Service, error) {
 		return nil, fmt.Errorf("failed to create admin client: %w", err)
 	}
 
```

Comment:
We are not checking for `EstablishConnectionEagerly` here. Maybe add a check and return early if it's disabled so that we won't test?

Reply:
I am not sure I understand..? There is a check if admin API is enabled [at the start of `NewService()` method](https://github.com/redpanda-data/console/blob/master/backend/pkg/redpanda/service.go#L36).

Reply:
But you introduced a new property `EstablishConnectionEagerly` which we had for the Kafka service as well. If this is enabled (default) we'll check if we can reach the API on startup. If we can't we will fail the startup process. This property is not used anywhere yet as far as I can see.

Reply:
`EstablishConnectionEagerly` is already present in `KafkaStartup` config, along with other backoff config properties. What I did was refactor these backoff properties into `ServiceStartupAttemptsOptions` config struct. Then `KafkaStartup` embeds the new backoff struct, along with `EstablishConnectionEagerly` property.
For `RedpandaAdminAPI` config struct I added just the new `ServiceStartupAttemptsOptions` as `startup` property. So the Admin API startup does not have `EstablishConnectionEagerly` because Admin API connection is kinda required on startup. We can't do it eagerly or lazily. Also I feel like `EstablishConnectionEagerly` doesn't belong in backoff options, but for scenarios where it is reasonable, we can just embed the backoff options like I did for Kafka service. Hope this makes sense.

Reply:
Ah correct, it's only in the Kafka config. Why do you think the admin api is required in contrast to the Kafka API? Clusters can totally work without any redpanda admin api connection at all, so I think it's actually less required than the Kafka API. I think we should add this property too, so that Console can also start (if desired) if the admin api is currently unreachable for whatever reason.

Reply:
We were saving the cluster version into the redpanda service struct... I had thought it was needed for something like endpoint compatibility checks... Upon further inspection we were not using the cluster version at all. I think I had added this previously when I was working on RBAC compatibility checks; but it was not needed. I have removed the cluster version from the struct, and added `EstablishConnectionEagerly` to the new generic `ServiceStartupAttemptsOptions`, which now is essentially the same as `KafkaStartup` was. `KafkaStartup` is removed and both Kafka and Redpanda configs now use `ServiceStartupAttemptsOptions`. In Redpanda service we perform the get cluster version connection check only if `EstablishConnectionEagerly` is `true` similar to the Kafka Service.